### ISSUE TITLE
"Prelude.(!!): index too large" error.

### ISSUE THREAD

**Issue by canavaroski90** [AtomLinter/linter-shellcheck#59]
linter-shellcheck enabled, linter installed and atom ide used for the test.
Ubuntu 14.04.03

Create a bash script (test.sh) with the following content;

```
#!/bin/bash
echo test1@"$tmpVar"
```

Remove the last double-quote ("). Atom throws following error:

```
Error: Prelude.(!!): index too large
    at parameters.exit (/home/user/.atom/packages/linter-shellcheck/node_modules/atom-linter/lib/helpers.js:47:27)
    at triggerExitCallback (/usr/share/atom/resources/app.asar/src/buffered-process.js:213:47)
    at /usr/share/atom/resources/app.asar/src/buffered-process.js:227:18
    at Socket.<anonymous> (/usr/share/atom/resources/app.asar/src/buffered-process.js:98:18)
    at emitOne (events.js:82:20)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
```

**Comment by Arcanemagus** [comment]
That error looks to be being forwarded on from `shellcheck` itself, does it throw the same error when run manually?

**Comment by canavaroski90** [comment]
I've tested following command in terminal.

`shellcheck test.sh`

It didn't throw any error.

**Comment by dgrant** [comment]
I'm seeing this as well.

**Comment by Arcanemagus** [comment]
How about this:
```shell
cat test.sh | shellcheck -f gcc -
```
(putting any custom parameters of yours before that last `-`)

**Comment by dgrant** [comment]
Problems seems to have gone now for me.

**Comment by Arcanemagus** [comment]
@canavaroski90 Are you still seeing this?

(The error is simply the contents of `stderr` after `shellcheck` exited, so whatever the issue is it has to do with `shellcheck`...)

**Comment by rugk** [comment]
I experienced this error when the syntax of the shell script was invalid. (using shellcheck from debian jessie repos)
@Arcanemagus If so can't you catch this exception and show a more useful error message or just hide the result?

**Comment by Arcanemagus** [comment]
@rugk That's all that `shellcheck` is returning, so that's as useful of a message as is possible from here.

**Comment by Arcanemagus** [comment]
@rugk are you still seeing this? Do you see it if you run the command in https://github.com/AtomLinter/linter-shellcheck/issues/59#issuecomment-209122861?

**Comment by rugk** [comment]
Yes, e.g. if I do not close a string (`"opened string, but no not close it`).

```

Error: Prelude.(!!): index too large
    at ChildProcess.<anonymous> (/home/<username>/.atom/packages/linter-jshint/node_modules/atom-linter/node_modules/sb-exec/lib/index.js:36:20)
    at emitTwo (events.js:100:13)
    at ChildProcess.emit (events.js:185:7)
    at maybeClose (internal/child_process.js:850:16)
    at Socket.<anonymous> (internal/child_process.js:323:11)
    at emitOne (events.js:90:13)
    at Socket.emit (events.js:182:7)
    at Pipe._onclose (net.js:477:12)
```

linter-spellcheck 1.3.0

When running the console command I also got this message, but most importantly I also got some kind of useful output (as it seems):
``` 
-:16:1: note: The mentioned parser error was in this simple command. [SC1009]
-:16:14: warning: Did you forget to close this double quoted string? [SC1078]
-:18:6: note: This is actually an end quote, but due to next char it looks suspect. [SC1079]
-:18:50: error: Couldn't parse this double quoted string. [SC1073]
Prelude.(!!): index too large
```

Especially - in my case with `SC1078` the actual issue has been found.

**Comment by Arcanemagus** [comment]
Looking into this a bit more, it looks like this is an error from Haskell (which ShellCheck appears to be written in), see [here](https://mail.haskell.org/pipermail/beginners/2013-April/011761.html).

I found https://github.com/koalaman/shellcheck/issues/228 which seems to be the same error that was _supposedly_ fixed, but seems not to be.

Does somebody who actually uses this linter want to report it over there and link back here for others to find it?

**Comment by canavaroski90** [comment]
Sorry for not being so active on this topic.

I still have the same problem and if you still insist on posting the same issue under ShellCheck section I can do it willingly.

Atom version 1.11.2
linter add-in 1.11.18
linter shellcheck add-in 1.4.1
bash shellcheck version 0.3.3

**Comment by Arcanemagus** [comment]
@canavaroski90 It's not that I'm "insisting" on it being posted there, it's just that there is absolutely nothing that can be done about it here so it would need to be reported over in ShellCheck if you want it fixed 😛.

**Comment by canavaroski90** [comment]
Yep, you're right.

Posting it over in ShellCheck section :+1:  :)

Thank you very much for your helping. :+1:

**Comment by Arcanemagus** [comment]
I'm going to close this issue as it has nothing to do with this project in the first place. Anyone coming across this in the future please see the above referenced issue!
